src/main/core/choo-choo-train.ts

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright IBM Corp. 2024, 2024
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import * as net from 'node:net'
+
+import configSchemaJson from '@ibm/telemetry-config-schema/config.schema.json' assert { type: 'json' }
+
+import { IbmTelemetry } from '../ibm-telemetry.js'
+import { Environment } from './environment.js'
+import { Loggable } from './log/loggable.js'
+import type { Logger } from './log/logger.js'
+import { Trace } from './log/trace.js'
+
+const MAX_RETRIES = 3
+const MAX_BACKLOG = 20
```

Comment:
Are we comfortable with this number? I feel like there might be projects instantiating more than 20 telemetry runs. In that case, would we lose out on the excedent?
Or do we expect them to clear out of the queue quick enough to allow the others in even if its 20+?

Reply:
@mattrosno Do you have a query which shows the maximum number of instrumented packages we've seen reported by a repo? Have we seen more than 20? We can bump this up to like 50 or something to add an extra safety net, but I don't want to go too high because each waiting connection will eat up some system resources.